python/pydantic_core/core_schema.py

```diff
@@ -659,6 +660,72 @@ def float_schema(
     )
 
 
+class DecimalSchema(TypedDict, total=False):
+    type: Required[Literal['decimal']]
+    gt: Union[int, Decimal]
+    ge: Union[int, Decimal]
+    lt: Union[int, Decimal]
+    le: Union[int, Decimal]
+    max_digits: int
+    decimal_places: int
+    multiple_of: Union[int, Decimal]
+    allow_inf_nan: bool  # whether 'NaN', '+inf', '-inf' should be forbidden. default: False
+    check_digits: bool  # FIXME document. default: False
+    strict: bool
+    ref: str
+    metadata: Any
+    serialization: SerSchema
+
+
+def decimal_schema(
+    *,
+    gt: int | Decimal | None = None,
+    ge: int | Decimal | None = None,
+    lt: int | Decimal | None = None,
+    le: int | Decimal | None = None,
+    max_digits: int | None = None,
+    decimal_places: int | None = None,
+    multiple_of: int | Decimal | None = None,
```

Comment:
looks like maybe this is missing `whole_digits`? I see it can be an error below but don't see a way to specify it

Reply:
After reading the validation code, it looks like whole_digits is basically `max_digits - decimal_places`. I didn't look closely but I wonder if it makes sense to allow these to be constrained separately? Not really a big deal I guess.

If this is essentially just copying the current python-side logic as faithfully as possible, I'm fine keeping it behaving as it currently does.

Reply:
Yep it's a copy of the existing Python logic, I'd be open to considering a change in a future PR, let's keep as-is for now.